client/my-sites/checkout/composite-checkout/use-get-thank-you-url.js

```diff
@@ -102,6 +103,13 @@ export function getThankYouPageUrl( {
 	debug( 'fallbackUrl is', fallbackUrl );
 
 	saveUrlToCookieIfEcomm( saveUrlToCookie, cart, fallbackUrl );
+
+	// If the user is making a purchase/upgrading within the editor,
+	// we want to return them back to the editor after the purchase is successful.
+	if ( isInEditor && ! hasEcommercePlan( cart ) ) {
+		saveUrlToCookie( window?.location.href );
```

Comment:
If possible, it would be great to add a test for this behavior in `client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js`